lib/dal/src/attribute/value.rs

```diff
@@ -2125,4 +2128,108 @@ impl AttributeValue {
     }
+    #[instrument(level = "info", skip_all)]
+    /// Get the morale equivalent of the prop path for a given attribute value id.
+    /// This includes the key/index in the path, unlike the path for a Prop which doesn't
+    /// include the key/index
+    pub async fn get_path_for_id(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+    ) -> AttributeValueResult<Option<String>> {
+        let mut parts = VecDeque::new();
+        let mut work_queue = VecDeque::from([attribute_value_id]);
+
+        while let Some(mut attribute_value_id) = work_queue.pop_front() {
+            let prop_name = match AttributeValue::prop_id_for_id(ctx, attribute_value_id).await {
+                Ok(prop) => Prop::get_by_id(ctx, prop).await?.name,
+                Err(_) => {
+                    // This means it's an input or output socket, which has no path and should just be empty string
+                    continue;
+                }
+            };
+
+            // get the parent to see if it's a Map or Array so we can add the necessary key/index to the prop path
+            if let Some(parent_attribute_value_id) =
+                AttributeValue::parent_attribute_value_id(ctx, attribute_value_id).await?
+            {
+                let parent_prop_id =
+                    AttributeValue::prop_id_for_id(ctx, parent_attribute_value_id).await?;
+                match Some(Prop::get_by_id(ctx, parent_prop_id).await?.kind) {
+                    Some(PropKind::Array) => {
+                        let order_of_thing_node = ctx
+                            .workspace_snapshot()?
+                            .incoming_sources_for_edge_weight_kind(
+                                attribute_value_id,
+                                EdgeWeightKindDiscriminants::Ordinal,
+                            )
+                            .await?
+                            .pop()
+                            .ok_or(AttributeValueError::NoOrderingNodeForAttributeValue(
+                                attribute_value_id,
+                            ))?;
+
+                        let order_node_weight = ctx
+                            .workspace_snapshot()?
+                            .get_node_weight(order_of_thing_node)
+                            .await?
+                            .get_ordering_node_weight()?;
+
+                        let order = order_node_weight.order();
+
+                        // Find the index of the item in the array to add to path
+                        let index = order
+                            .iter()
+                            .position(|&id| id == attribute_value_id.into())
+                            .ok_or(AttributeValueError::ParentAttributeValueMissing(
+                                attribute_value_id,
+                            ))?;
+                        let name = format!("{0}[{1}]", prop_name, &index.to_string());
+                        parts.push_front(name.to_owned());
```

Comment:
What is being assembled here? It looks like the path for display purposes in the frontend.

Reply:
That is correct. This function is supposed to assemble a path to the attribute value, for display in the debugging panel. So if its an array item, it would be something like `/root/domain/things/thing[0]` with `0` being the index of the item, `thing` being the prop name for entry and `things` the prop name for the array.

For a map its the same thing but with key instead of index

Reply:
Gotcha. Is there a reason why we wouldn't use `PropPath` instead of strings? Is it because it doesn't work with array items?

Reply:
Exactly. Prop paths are different from the attribute value paths this creates since we show keys and indices

Reply:
I see!! String manipulation makes me nervous... I wonder if an `AttributeValuePath` type entity would be useful.

Reply:
Yeah I debated making an `AttributeValuePath` entity, because I was honestly surprised we didn't have this yet. But it felt strange to formalize it for just this debug view and I figured if we end up needing it later we can always formalize it then and replace this implementation with that one.

Reply:
I won't be a blocker here, but I would personally formalize it because there may be a day 2 bug or feature extension here that relies on using/changing/extending the string manipulation.

It's a nit though.